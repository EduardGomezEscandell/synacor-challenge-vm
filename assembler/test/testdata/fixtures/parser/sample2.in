set r0 text
loop:
    rmem r8 r0
    out r8

    add r0 r0 1

    eq r1 r0 endtext
    jt r1 loop
halt

text:
    "Hello, world!"
endtext: