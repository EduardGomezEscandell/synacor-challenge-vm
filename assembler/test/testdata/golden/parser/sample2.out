<Start>
  <P>
    <I>
      <VERB set>
      <R>
        <REGISTER 0>
      <W>
        <TAG_REF text>
    <EOL>
    <P>
      <T>
        <TAG_DECL loop>
      <EOL>
      <P>
        <I>
          <VERB rmem>
          <R>
            <REGISTER 8>
          <W>
            <REGISTER 0>
        <EOL>
        <P>
          <I>
            <VERB out>
            <W>
              <REGISTER 8>
          <EOL>
          <P>
            <I>
              <VERB add>
              <R>
                <REGISTER 0>
              <W>
                <REGISTER 0>
              <W>
                <NUMBER 1>
            <EOL>
            <P>
              <I>
                <VERB eq>
                <R>
                  <REGISTER 1>
                <W>
                  <REGISTER 0>
                <W>
                  <TAG_REF endtext>
              <EOL>
              <P>
                <I>
                  <VERB jt>
                  <W>
                    <REGISTER 1>
                  <W>
                    <TAG_REF loop>
                <EOL>
                <P>
                  <I>
                    <VERB halt>
                  <EOL>
                  <P>
                    <T>
                      <TAG_DECL text>
                    <EOL>
                    <P>
                      <D>
                        <STRING Hello, world!>
                        <D>
                      <EOL>
                      <P>
                        <T>
                          <TAG_DECL endtext>
                        <EOL>
                        <P>
                          <END>
  <END>
